test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -72,6 +72,50 @@ def test_server_error
     end
   end
 
+  def test_prometheus_metrics
+    setup_gce_metadata_stubs
+    [
+      # Single successful request.
+      [false, 0, 1, 1, [1, 0, 1, 0]],
+      # Several successful requests.
+      [false, 0, 2, 1, [2, 0, 2, 0]],
+      # Single successful request with several entries.
+      [false, 0, 1, 2, [1, 0, 2, 0]],
+      # Single failed request that causes logs to be dropped.
+      [true, 16, 1, 1, [0, 1, 0, 1]],
```

Comment:
Technically we should be using `GRPC::BadStatus::UNAUTHENTICATED`, etc, instead of these int literals. But that's beyond the scope of this PR...